commit ae02874a7304f43ab932f000e23566828a52e89d (refs/changes/71/1471/5)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-02-08T02:20:42+00:00 (2 years, 8 months ago)
    
    joyent/node-cueball#91 deps: restify-clients isn't used, use uuid instead of node-uuid
    Reviewed by: Alex Wilson <alex.wilson@joyent.com>

diff --git a/CHANGES.adoc b/CHANGES.adoc
index 91ba027..9495c36 100644
--- a/CHANGES.adoc
+++ b/CHANGES.adoc
@@ -21,6 +21,8 @@ Bugs fixed:
  - #85 documentation improvements and re-organization
  - #26 clarified that programs that wish to exit cleanly without calling
    `process.exit()` must call `.stop()` on agents or pools.
+ - #90 ensure cueball works with node 0.10
+ - #88 correct a bug that cause a crash bringing up a backend after pool failure
 
 ### v2.0.1
 
diff --git a/lib/connection-fsm.js b/lib/connection-fsm.js
index 2fa697c..8677860 100644
--- a/lib/connection-fsm.js
+++ b/lib/connection-fsm.js
@@ -3,7 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  *
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 module.exports = {
@@ -21,7 +21,6 @@ const mod_vasync = require('vasync');
 const mod_verror = require('verror');
 const mod_bunyan = require('bunyan');
 const mod_resolver = require('./resolver');
-const mod_uuid = require('node-uuid');
 const mod_errors = require('./errors');
 
 const mod_monitor = require('./pool-monitor');
diff --git a/lib/pool.js b/lib/pool.js
index 8bcc13e..58a8cb3 100644
--- a/lib/pool.js
+++ b/lib/pool.js
@@ -3,7 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  *
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 module.exports = {
@@ -19,7 +19,7 @@ const mod_utils = require('./utils');
 const mod_vasync = require('vasync');
 const mod_bunyan = require('bunyan');
 const mod_resolver = require('./resolver');
-const mod_uuid = require('node-uuid');
+const mod_uuid = require('uuid');
 const mod_errors = require('./errors');
 
 const mod_monitor = require('./pool-monitor');
diff --git a/lib/set.js b/lib/set.js
index 40d5af2..bbb2d0c 100644
--- a/lib/set.js
+++ b/lib/set.js
@@ -3,7 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  *
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright 2017 Joyent, Inc.
  */
 
 module.exports = {
@@ -19,7 +19,7 @@ const mod_utils = require('./utils');
 const mod_vasync = require('vasync');
 const mod_bunyan = require('bunyan');
 const mod_resolver = require('./resolver');
-const mod_uuid = require('node-uuid');
+const mod_uuid = require('uuid');
 const mod_errors = require('./errors');
 const mod_monitor = require('./pool-monitor');
 
diff --git a/package.json b/package.json
index 50e34b5..22b086c 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cueball",
   "version": "2.1.0",
-  "description": "",
+  "description": "manage a pool of connections to a multi-node service where nodes are listed in DNS",
   "main": "lib/index.js",
   "dependencies": {
     "assert-plus": ">=1.0.0 <2.0.0",
@@ -11,9 +11,8 @@
     "ipaddr.js": ">=1.1.0 <2.0.0",
     "mooremachine": ">=2.1.0 <3.0.0",
     "mname-client": ">=0.5.0 <0.6.0",
-    "node-uuid": ">=1.4.7 <2.0.0",
+    "uuid": ">=3.0.1 <4.0.0",
     "posix-getopt": ">=1.2.0 <2.0.0",
-    "restify-clients": ">=1.1.2 <2.0.0",
     "vasync": ">=1.6.3 <2.0.0",
     "verror": ">=1.6.1 <2.0.0"
   },
@@ -27,6 +26,9 @@
     "json": ">=9.0.3 <10.0.0",
     "mname": ">=1.0.8 <2.0.0"
   },
+  "engines": {
+    "node": ">=0.10.0"
+  },
   "scripts": {
     "test": "tape test/*.test.js"
   },
